core/internal/api/credentials_test.go

```diff
@@ -1,22 +1,28 @@
 package api_test
 
 import (
+	"encoding/json"
 	"net/http"
+	"os"
 	"testing"
+	"time"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	"github.com/wandb/wandb/core/internal/api"
+	"github.com/wandb/wandb/core/internal/apitest"
+	"github.com/wandb/wandb/core/internal/observability"
 	wbsettings "github.com/wandb/wandb/core/internal/settings"
 	spb "github.com/wandb/wandb/core/pkg/service_go_proto"
+	"golang.org/x/sync/errgroup"
 	"google.golang.org/protobuf/types/known/wrapperspb"
 )
 
 func TestNewAPIKeyCredentialProvider(t *testing.T) {
 	settings := wbsettings.From(&spb.Settings{
 		ApiKey: &wrapperspb.StringValue{Value: "test-api-key"},
 	})
-	credentialProvider, err := api.NewCredentialProvider(settings)
+	credentialProvider, err := api.NewCredentialProvider(settings, observability.NewNoOpLogger().Logger)
```

Comment:
```suggestion
	credentialProvider, err := api.NewCredentialProvider(
		settings,
		observability.NewNoOpLogger().Logger,
	)
```

![Screenshot 2024-11-14 at 5 43 42 PM](https://github.com/user-attachments/assets/c0917dda-41e8-47ca-8832-ed029989420c)